org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventPredicates.java

```diff
@@ -0,0 +1,53 @@
+/*******************************************************************************
+ * Copyright (c) Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *******************************************************************************/
+
+package org.osgi.test.assertj.bundleevent;
+
+import java.util.function.Predicate;
+
+import org.osgi.framework.BundleEvent;
+
+/**
+ * The Interface BundleEventPredicates.
+ */
+public class BundleEventPredicates {
+
+	private BundleEventPredicates() {}
+
+	// BundleEvents
+
+	/**
+	 * Returns a predicate that tests if the object Bundle event.
+	 *
+	 * @return the predicate
+	 */
+	public static Predicate<Object> bundleEvent() {
```

Comment:
I think we may want to declare this as `public static <T> Predicate<T> bundleEvent()`. This would then allow someone to chain predicates and not lose the type.

```java
Predicate<BundleEvent> p = bundleEvent().and(type(BundleEvent.INSTALLED));
```

But I am not sure how useful this would be in practice.

Reply:
fixed